### DISCUSSION TITLE
Client#publish(nil)

### DISCUSSION THREAD

**Discussion by grncdr** [RailsEventStore/rails_event_store#1269] [Q&A]
Not sure if this is a bug, but I noticed today that `event_store.publish(nil)` does not raise an error.

Is this intentional behavior? Personally I found it very surprising.

---

Context is that I had some code like:

```rb
do_something(record)
event = choose_event(record)
event_store.publish(event, stream_name: "some_stream")
do_something_else(record)
```

and `#choose_event` is something like this:

```rb
def choose_event(record)
  case record.status
  when 'foo'
     SomeEvent.new(...)
  when 'bar'
     SomeOtherEvent.new(...)
  end
end
```

Clearly that `case` code should have had an `else` to ensure it was exhaustive, but it didn't, and so the nil event was "published" instead. Made for a funny debug session when the logs showed `do_something` and `do_something_else`, but nothing for `event_store.publish(...)`.

**Comment by jandudulski** [comment]
I believe it's by coincidence - there is a conversion of argument into an array and `Array(nil)` returns an empty array (which is acceptable input too). But I'm not sure if there should be a guard against this case or not :) @pawelpacana any opinion?

**Reply by mostlyobvious** [reply]
It would make sense to me to guard against it in the Client facade.

**Reply by jandudulski** [reply]
https://github.com/RailsEventStore/rails_event_store/pull/1354 PR ready

**Reply by mostlyobvious** [reply]
Had to reschedule introduction of this to next major version https://github.com/RailsEventStore/rails_event_store/pull/1466

While it wasn't specified, someone could have already assumed existing observed behaviour.
